src/OSPSuite.Core/Domain/ModelCoreSimulationExtensions.cs

```diff
@@ -39,7 +39,7 @@ private static IReadOnlyList<IContainer> allApplicationsForMolecule(IModelCoreSi
          var allApplications = applicationEventGroup.SelectMany(x => x.GetAllChildren<IContainer>(c => c.ContainerType == ContainerType.Application))
             .ToList();
 
-         return getApplicationsForAppliedAncestorMolecule(simulation.Reactions, moleculeName, allApplications, new List<string>());
+         return getApplicationsForAppliedAncestorMolecule(simulation.Reactions.SelectMany(x=>x), moleculeName, allApplications, new List<string>());
```

Comment:
Why do we need this:` .SelectMany(x=>x)` ??

Reply:
Because we go from Enumerabtion of BB to enumeration of reactions. SelectMany flattens it all

Reply:
ok

Reply:
not sure we need this. Right now I am combining all over all . This is probably not what we want. Just want to make the tests run again and they break everything again :)